components/function-controller/internal/controllers/serverless/function_controller.go

```diff
@@ -48,7 +48,9 @@ func (r *FunctionReconciler) envsEqual(existing, expected []corev1.EnvVar) bool
 	for key, value := range existing {
 		expectedValue := expected[key]
 
-		if expectedValue.Name != value.Name || expectedValue.Value != value.Value || expectedValue.ValueFrom != value.ValueFrom { // valueFrom check is by reference
+
+
+		if expectedValue.Name != value.Name || expectedValue.Value != value.Value || expectedValue.ValueFrom.String() != value.ValueFrom.String() { // valueFrom check is by string representation
```

Comment:
If think that before string comparison, we should check existing of pointers.

Reply:
it's already done by String() method